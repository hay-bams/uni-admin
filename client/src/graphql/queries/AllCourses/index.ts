import { gql } from '@apollo/client';

export const All_COURSES = gql`
  query AllCourses {
    allCourses {
      id
      name
      totalSeats
      status
    }
  }
`;
