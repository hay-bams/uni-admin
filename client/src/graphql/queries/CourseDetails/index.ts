import { gql } from '@apollo/client';

export const COURSE_DETAILS = gql`
  query courseDetails($courseId: ID!) {
    courseDetails(courseId: $courseId) {
      id
      name
      status
      totalSeats
    }
  }
`;
