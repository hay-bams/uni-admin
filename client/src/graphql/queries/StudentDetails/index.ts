import { gql } from "@apollo/client";

export const STUDENT_DETAILS = gql`
  query studentDetails ($id: ID!) {
    studentDetails(id: $id) {
      id
      name
      courses {
        id
        name
        totalSeats
        status
      }
    }
  }
`;
