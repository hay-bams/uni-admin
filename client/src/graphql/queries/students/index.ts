import { gql } from '@apollo/client';

export const STUDENT = gql`
  query Students($all: String, $page: Int, $limit: Int) {
    students(all: $all, page: $page, limit: $limit) {
      total
      results {
        id
        studentID
        dob
        status
        name
        email
        country
      }
    }
  }
`;
