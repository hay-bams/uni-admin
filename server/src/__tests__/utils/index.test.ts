/* eslint-disable @typescript-eslint/no-non-null-assertion */
import jwt from 'jsonwebtoken';
import { COURSES_DATA, USER_DATA } from '../test_data';
import { createTestServer } from '../helpers';
import { TEST_All_COURSES_QUERY } from '../queries/queries';

describe('Utils', () => {
  test('should return an error if user is not authenticated', async () => {
    const { query } = createTestServer({
      db: {
        courses: {
          find: jest.fn(() => ({
            count: jest.fn(() => 4),
            toArray: jest.fn(() => COURSES_DATA),
          })),
        },
        users: {
          findOne: jest.fn(() => USER_DATA[0]),
        },
      },
      req: {
        get: jest.fn(() => 'skskkdkdk'),
        signedCookies: {
          admin: '',
        },
      },
      res: {
        clearCookie: jest.fn(),
      },
    });

    jest.spyOn(jwt, 'verify').mockImplementation(
      jest.fn(() => ({
        admin: USER_DATA[0]._id,
      }))
    );

    const res = await query({ query: TEST_All_COURSES_QUERY });

    expect(res.errors?.length).toBeGreaterThan(0);
    expect(res!.errors![0]).toHaveProperty('message');
    expect(res).toMatchSnapshot();
  });

  test('should return an error if user is not found in the database', async () => {
    const { query } = createTestServer({
      db: {
        courses: {
          find: jest.fn(() => ({
            count: jest.fn(() => 4),
            toArray: jest.fn(() => COURSES_DATA),
          })),
        },
        users: {
          findOne: jest.fn(() => null),
        },
      },
      req: {
        get: jest.fn(() => 'skskkdkdk'),
        signedCookies: {
          admin: USER_DATA[0],
        },
      },
      res: {
        clearCookie: jest.fn(),
      },
    });

    jest.spyOn(jwt, 'verify').mockImplementation(
      jest.fn(() => ({
        admin: USER_DATA[0]._id,
      }))
    );

    const res = await query({ query: TEST_All_COURSES_QUERY });

    expect(res.errors?.length).toBeGreaterThan(0);
    expect(res!.errors![0]).toHaveProperty('message');
    expect(res).toMatchSnapshot();
  });
});
