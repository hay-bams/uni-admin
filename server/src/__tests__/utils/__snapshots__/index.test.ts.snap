// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils should return an error if user is not authenticated 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Something went wrong: TypeError: cursor.skip is not a function],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Utils should return an error if user is not found in the database 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Something went wrong: TypeError: cursor.skip is not a function],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
